src/layer/x86/instancenorm_x86.cpp

```diff
@@ -0,0 +1,217 @@
+        __m128 vhigh_sq = _mm256_extractf128_ps(_sqsum_avx, 1);
+        __m128 hsqsum_128 = _mm_add_ps(vlow_sq, vhigh_sq);
+        hsqsum_128 = _mm_hadd_ps(hsqsum_128, hsqsum_128);
+        hsqsum_128 = _mm_hadd_ps(hsqsum_128, hsqsum_128);
+        sqsum += _mm_cvtss_f32(hsqsum_128);
+#endif // __AVX__
+        __m128 _sqsum = _mm_setzero_ps();
+        __m128 _mean = _mm_set1_ps(mean);
+
+        for (; i <= size - 4; i += 4)
+        {
+            __m128 _p = _mm_loadu_ps(ptr + i);
+            __m128 _diff = _mm_sub_ps(_p, _mean);
+            _sqsum = _mm_add_ps(_sqsum, _mm_mul_ps(_diff, _diff));
+        }
+
+        __m128 h_sqsum_tmp = _sqsum;
+        h_sqsum_tmp = _mm_add_ps(h_sqsum_tmp, _mm_movehl_ps(h_sqsum_tmp, h_sqsum_tmp));
+        h_sqsum_tmp = _mm_add_ss(h_sqsum_tmp, _mm_shuffle_ps(h_sqsum_tmp, h_sqsum_tmp, _MM_SHUFFLE(1, 1, 1, 1)));
+        sqsum += _mm_cvtss_f32(h_sqsum_tmp);
+#endif // __SSE2__
+
+        for (; i < size; i++)
+        {
+            tmp = ptr[i] - mean;
+            sqsum += tmp * tmp;
+        }
+        float var = sqsum / size;
+        // the var maybe minus due to accuracy
+        //float var = sqsum / size - mean * mean;
+
+        float a;
+        float b;
+        if (affine)
+        {
+            float gamma = gamma_data[q];
+            float beta = beta_data[q];
+
+            a = gamma / (sqrtf(var + eps));
+            b = -mean * a + beta;
+        }
+        else
+        {
+            a = 1.f / (sqrtf(var + eps));
+            b = -mean * a;
+        }
+
+        i = 0;
+
+#if __SSE2__
+#if __AVX__
+#if __AVX512F__
+        __m512 _va_avx512 = _mm512_set1_ps(a);
+        __m512 _vb_avx512 = _mm512_set1_ps(b);
+
+        for (; i <= size - 16; i += 16)
+        {
+            __m512 _p = _mm512_loadu_ps(ptr + i);
+            _p = _mm512_fmadd_ps(_p, _va_avx512, _vb_avx512); // Fused Multiply-Add
+            _mm512_storeu_ps(ptr + i, _p);
+        }
+#endif // __AVX512F__
+        __m256 _va_avx = _mm256_set1_ps(a);
+        __m256 _vb_vax = _mm256_set1_ps(b);
+
+        for (; i <= size - 8; i += 8)
+        {
+            __m256 _p = _mm256_loadu_ps(ptr + i);
+#if defined(__FMA__) // Check for FMA support (usually with AVX2)
+            _p = _mm256_fmadd_ps(_p, _va_avx, _vb_vax);
+#else
+            _p = _mm256_mul_ps(_p, _va_avx);
+            _p = _mm256_add_ps(_p, _vb_vax);
+#endif
+            _mm256_storeu_ps(ptr + i, _p);
+        }
+#endif // __AVX__
+        __m128 _va = _mm_set1_ps(a);
+        __m128 _vb = _mm_set1_ps(b);
+
+        for (; i <= size - 4; i += 4)
+        {
+            __m128 _p = _mm_loadu_ps(ptr + i);
+            // SSE2 does not have FMA instructions directly; FMA is part of FMA3 extension.
+            // Compilers might fuse mul+add if target architecture supports it (e.g. compiling with -mfma).
+            _p = _mm_mul_ps(_p, _va);
+            _p = _mm_add_ps(_p, _vb);
+            _mm_storeu_ps(ptr + i, _p);
```

Comment:
there is fma wrapper `_mm_comp_fmadd_ps` `_mm256_comp_fmadd_ps` in x86_usability.h